CONTRIBUTING.md

```diff
@@ -0,0 +1,258 @@
+The templates are great, and high-quality issues really do help us!
+Do your best to search for duplicate issues, but if you're unsure, open a new issue and link to other related issues within.
+
+And once you've made something that you're proud of, feel free to drop a link, video or screenshot in `#showcase` on Discord!
+If you release a game on [itch.io](https://itch.io/games/tag-bevy) we'd be thrilled if you tagged it with `bevy`.
+
+### Teaching others
+
+Bevy is still very young, and light on documentation, tutorials and accumulated expertise.
+By teaching others and helping them with their issues, you can learn the engine yourself and make our community better!
+
+Some of the best ways to do this are:
+
+* answering questions on [Discord](https://discord.gg/bevy), [GitHub Discussions](https://github.com/bevyengine/bevy/discussions), [reddit](https://www.reddit.com/r/bevy) or [StackOverflow](https://stackoverflow.com/questions/tagged/bevy).
+* writing tutorials, guides and other informal documentation and sharing it on [awesome-bevy](https://github.com/bevyengine/awesome-bevy)
+* streaming, writing blog posts about creating your game or creating videos. Share these in the `#devlogs` channel on Discord!
+
+### Writing plugins
+
+If you're interested in contributing to the community ecosystem in a way that doesn't make sense as part of the core engine,
+feel free to write a plugin or crate for Bevy!
+
+Any useful, reusable and non-trivial pieces of functionality that you've made serves as a good candidate for a plugin.
+If it's closer to a snippet or design pattern, you may want to share it with the community on Discord, Reddit or GitHub Discussions instead.
+
+Check out our [plugin guidelines](https://github.com/bevyengine/bevy/blob/main/docs/plugins_guidelines.md) for helpful tips and norms.
+
+### Fixing bugs
+
+Bugs in Bevy (or our website / books) are filed on the issue tracker corresponding to where they're found, using the [`bug`](https://github.com/bevyengine/bevy/issues?q=is%3Aissue+is%3Aopen+label%3Abug) label.
+
+If you're looking for an easy place to start, take a look at the [`good-first-issue`](https://github.com/bevyengine/bevy/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) tag, and feel free to ask questions on the issue in question or on Discord.
+You don't need anyone's permission to try fixing a bug or adding a simple feature, but stating that you'd like to tackle an issue can be helpful to avoid duplicated work.
+
+When you make a pull request that fixes an issue, include a line that says `Fixes #X` (or "Closes"), where `X` is the issue number.
+This will cause the issue in question to be closed when your PR is merged.
+
+General improvements to code quality are also welcome!
+Bevy can always be safer, better tested, and more idiomatic.
+
+### Writing docs
+
+Like every other large, rapidly developing open source library you've ever used, Bevy's documentation could stand to be improved.
```

Comment:
rapidly-developing open-source library ~~you've ever used~~, Bevy's documentation can always use improvement.